src/main/groovy/org/shipkit/internal/gradle/plugin/tasks/PluginValidator.java

```diff
@@ -1,95 +1,120 @@
 package org.shipkit.internal.gradle.plugin.tasks;
 
 import org.gradle.api.GradleException;
+import org.gradle.api.Project;
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.SourceSet;
+import org.shipkit.internal.gradle.util.JavaPluginUtil;
+import org.shipkit.internal.gradle.util.StringUtil;
 
 import java.io.File;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
+import static org.shipkit.internal.gradle.plugin.tasks.PluginUtil.getImplementationClass;
 import static org.shipkit.internal.gradle.plugin.tasks.PluginUtil.DOT_PROPERTIES;
 
 
 public class PluginValidator {
```

Comment:
No unit test for this class?

Reply:
@wwilk, you are probably right and more balanced coverage, with more explicit unit tests would drive slightly cleaner codebase.

PluginValidationPluginTest is sort of an integration test that covers this functionality. So we have the coverage, but necessarily as an explicit unit test to PluginValidator. I'd say that we move on, we don't necessarily have to have 1:1 class to test class in the codebase for every functionality. It's important that we have coverage. Coarse-grained tests are sometimes useful because refactorings are easier :) Let's move on and learn on the way if this way of writing tests works in our favor.